src/data/upgrade.ts

```diff
@@ -0,0 +1,15 @@
+import { Event, UpgradeData } from "@clarity-types/data";
+import config from "@src/core/config";
+import encode from "@src/data/encode";
+
+export let data: UpgradeData = null;
+
+export function reset(): void {
+    data = null;
+}
+
+export function upgrade(key: string): void {
```

Comment:
>upgrade [](start = 16, length = 7)

Who calls this function and when? What do we do when we upgrade? A bit of a higher level design comment would be helpful.

Reply:
Good point, done.

---
In reply to: [368681354](https://github.com/microsoft/clarity-js/pull/166#discussion_r368681354) [](ancestors = 368681354)